Sources/StreamChat/Config/StreamRuntimeCheck.swift

```diff
@@ -29,9 +24,7 @@ public enum StreamRuntimeCheck {
     ///
     ///  Returns true if the maximum depth of relationships to fetch when performing a mapping is not yet met
     static func _canFetchRelationship(currentDepth: Int) -> Bool {
-        guard _isBackgroundMappingEnabled else { return true }
-
-        return currentDepth <= _backgroundMappingRelationshipsMaxDepth
+        currentDepth <= _backgroundMappingRelationshipsMaxDepth
```

Comment:
Do we still need this? 🤔

Reply:
I don't know the background of it but feels like it is just limiting how much work the DB fetch is doing. Not sure about the side-effects. I think I would like to do this in a separate PR and then spend some time testing it.
Because we throw `RecursionError` then I would like to take it separately.

Reply:
Yes lets handle it in another PR 👍